engine/lite/components/kafka-tests/src/test/scala/pl/touk/nussknacker/engine/lite/components/UniversalSourceJsonFunctionalTest.scala

```diff
@@ -12,60 +12,81 @@ import org.scalatest.matchers.should.Matchers
-
-      //trimming not nullable and not required field when we sent null / empty object
-      (oConfig(InputEmptyObject, schemaObjNull, schemaObjNull), valid(obj())),
-      (oConfig(InputEmptyObject, schemaObjNull, schemaObjNull, SpecialSpELElement("#input.field")), oValid(Null)),
-      (oConfig(Null, schemaObjNull, schemaObjNull), oValid(Null)),
-      (oConfig(Null, schemaObjNull, schemaObjNull, SpecialSpELElement("#input.field")), oValid(Null)),
+      (sConfig(sampleInteger, schemaIntegerRange, schemaInteger), valid(fromInt(1))),
+      (sConfig(fromLong(Integer.MAX_VALUE), schemaIntegerRange, schemaIntegerRange), valid(fromInt(Integer.MAX_VALUE))),
+    )
 
-      (oConfig(InputEmptyObject, schemaObjString, schemaObjString), valid(obj())),
-      (oConfig(InputEmptyObject, schemaObjString, schemaObjString, SpecialSpELElement("#input.field")), valid(obj())),
+    forAll(testData) { (config: ScenarioConfig, expected: Validated[_, RunResult[_]]) =>
+      val results = runWithValueResults(config)
+      results shouldBe expected
+    }
+  }
 
-      (oConfig(InputEmptyObject, schemaObjUnionNullString, schemaObjUnionNullString, SpecialSpELElement("#input.field")), oValid(Null)),
-      (oConfig(InputEmptyObject, schemaObjUnionNullString, schemaObjUnionNullString), valid(obj())),
-      (oConfig(Null, schemaObjUnionNullString, schemaObjUnionNullString), oValid(Null)),
-      (oConfig(Null, schemaObjUnionNullString, schemaObjUnionNullString, SpecialSpELElement("#input.field")), oValid(Null)),
+  // TODO: add more tests for handling objects..
+  test("should test end to end kafka json data at sink and source / handling objects..") {
+    val testData = Table(
+      ("config", "result"),
+      (oConfig(sampleObjFirstLastName, schemaObjObjFirstLastNameRequired, schemaObjObjFirstLastNameRequired), oValid(sampleObjFirstLastName)),
+      (oConfig(sampleInteger, schemaObjInteger, schemaObjObjFirstLastNameRequired, sampleSpELFirstLastName), oValid(sampleObjFirstLastName)),
+
+      //Additional fields turn on
+      (oConfig(sampleMapAny, schemaObjMapAny, schemaObjMapAny), oValid(sampleMapAny)),
+      (oConfig(sampleInteger, schemaObjInteger, schemaObjMapAny, sampleMapSpELAny), oValid(sampleMapAny)),
+      (oConfig(sampleObjFirstLastName, schemaObjObjFirstLastNameRequired, schemaObjMapAny), oValid(sampleObjFirstLastName)),
+      (oConfig(sampleInteger, schemaObjInteger, schemaObjMapAny, sampleSpELFirstLastName), oValid(sampleObjFirstLastName)),
+
+      (oConfig(sampleMapInteger, schemaObjMapInteger, schemaObjMapInteger), oValid(sampleMapInteger)),
+      (oConfig(sampleInteger, schemaObjInteger, schemaObjMapInteger, sampleMapSpELInteger), oValid(sampleMapInteger)),
+      (sConfig(sampleInteger, schemaInteger, schemaObjString, Map("redundant" -> "red")), invalid(Nil, List("field"), List("redundant"))),
```

Comment:
can we also test situation where input schema has addionalProperitres = false, and we pass redudant paraemeter on input?

Reply:
That think is tested on Deserialization? Here we test validation schema.

Reply:
But it's probably unit-tested and here we have Functional tests. Maybe it is worth to have this tested here as well?

Reply:
Added some tests